### PR TITLE
Add querystring_search get method

### PR DESCRIPTION
**Pull request by robgietema**
See #1252 for details

### CODE REVIEW

**Review by sneridagh — APPROVED** [review]
LGTM

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
@robgietema Now that I realise, I think the only thing missing is a bit on documentation.

**Comment by robgietema** [comment]
> @robgietema Now that I realise, I think the only thing missing is a bit on documentation.

Agreed, I did add them but it seems it got lost while merging. I've added the docs again.

File: docs/source/endpoints/querystringsearch.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 myst:
   html_meta:
-    "description": "The @querystring-search endpoint returns search results that can be filtered on search criteria."
-    "property=og:description": "The @querystring-search endpoint returns search results that can be filtered on search criteria."
-    "property=og:title": "Querystring Search"
-    "keywords": "Plone, plone.restapi, REST, API, Querystring, Search"
+    'description': 'The @querystring-search endpoint returns search results that can be filtered on search criteria.'
+    'property=og:description': 'The @querystring-search endpoint returns search results that can be filtered on search criteria.'
+    'property=og:title': 'Querystring Search'
+    'keywords': 'Plone, plone.restapi, REST, API, Querystring, Search'
```
Comment (stevepiercy) [inline docs/source/endpoints/querystringsearch.md:7]: Please revert the above changes, and adjust your editor settings.

File: docs/source/endpoints/querystringsearch.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 ---
 
 # Querystring Search
 
 The `@querystring-search` endpoint returns search results that can be filtered on search criteria.
 
-Call the `/@querystring-search` endpoint with a `POST` request and a query in the request body:
+Call the `/@querystring-search` endpoint with a `POST` or a `GET`. When using the `POST` request you provide a query in the request body:
```
Comment (stevepiercy) [inline docs/source/endpoints/querystringsearch.md:14]:
```suggestion
Call the `/@querystring-search` endpoint with either a `POST` or `GET` request.

When using the `POST` request, provide a query in the request body:
```

File: docs/source/endpoints/querystringsearch.md [hunk 3]
```diff
@@ -24,6 +24,19 @@ The server will respond with the results that are filtered based on the query yo
 :language: http
 ```
 
+When using the `GET` request you provide the query as a json urlencoded string as a parameter.
```
Comment (stevepiercy) [inline docs/source/endpoints/querystringsearch.md:27]:
```suggestion
When using the `GET` request, provide the query as a JSON URL-encoded string as a parameter.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Minor tweaks to the docs.

**Review by davisagli — COMMENTED** [review]
LGTM, aside from the doc tweaks that Steve requested

File: src/plone/restapi/services/querystringsearch/get.py [hunk 4]
```diff
@@ -14,11 +17,14 @@
     SUPPORT_NOT_UUID_QUERIES = False
 
 
-class QuerystringSearchPost(Service):
```
Comment (tisto) [inline src/plone/restapi/services/querystringsearch/get.py:17]: This is a breaking change I guess. Not saying we can not do a breaking. Just mentioning it here...

Comment (tisto) [reply src/plone/restapi/services/querystringsearch/get.py:17]: I should scroll down before writing a comment. Please ignore me and sorry for the noise. :)

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@robgietema LGTM! Awesome that we can just add this feature without the need for a new major version. :)

When reading the docs, I was wondering if we could be more explicit about how URL encoding works. We could either add one or two lines on how to do the conversion (e.g. in JavaScript) works or link to existing documentation (risky because it could break in the future):

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI?retiredLocale=de

What do you think? @stevepiercy opinions regarding external links vs inline docs?

I saw that the Volto part (https://github.com/plone/volto/pull/4658) already has been merged. I am around to making a release anytime today.

**Comment by stevepiercy** [comment]
Whoops, I just saw your comment @tisto after approving the docs.

I agree with your idea. An entry in the Glossary would be preferred, using a standard definition, JavaScript method to do the encoding, and a link to MDN for details and demos. Keeps the main docs clean, while providing a reference that we can use here and other parts of the docs.

I also noticed that we use "URL-encoded", but should it be "URI-encoded"?

**Pull request merged**